ddtrace/internal/_encoding.pyx

```diff
@@ -1,6 +1,6 @@
 from cpython cimport *
 from cpython.bytearray cimport PyByteArray_Check
-import struct
+from libc cimport stdint
 import threading
 
 from ..constants import ORIGIN_KEY
```

Comment:
one further small optimisation would be to "unbox" this Python string to get hold of the underlying `char *` buffer so that we can call `msgpack_pack_raw*` directly on it instead of going through the heavier `pack_text`.

Reply:
I'm thinking we just remove this import entirely and replace the ``pack_text(&self.pk, ORIGIN_KEY)`` with:
```python
ret = pack_bytes(&self.pk, <char *> b"_dd.origin", 10)
```
If we're doing the same for `span_id`, `trace_id`, `parent_id`, `service`, `resource`, `name`, `error`, `start`, `duration`, `type`, `meta`, `metrics`, I don't see why we shouldn't also just hard code the bytestring for dd_origin as well. @brettlangdon @P403n1x87 What do you think?

Reply:
👍🏻 this is a good change to make

Reply:
Agreed. Presumably `_dd.origin` is unlikely to change, but in case it does we wouldn't have to worry about updating the string here too if we used the import.